src/payment/strategies/sdk/strategy.ts

```diff
@@ -0,0 +1,155 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { handler } from './handler';
+import { initializers, PaymentProcessor } from './initializers';
+
+/*
+    This strategy would be used for all PPSDK methods
+
+    Key points:
+        - The use of a "PPSDK Response Handler", which would
+            (1) discern different code paths based on success/failure/error/continue
+                responses from our checkout/processing endpoint (received via HostedFormFactory above)
+            (2) not return control to the strategy until all continue steps are exhausted
+            (3) attempt to handle any recoverable failure responses(?)
+            (4) translate error and un-recoverable failure responses into Checkout SDK errors
+                (e.g. those from src/common/error/errors/index.ts)
+                before throwing
+            (5) Delegate and await the resolving of relevant(?) continue steps to checkout-js
+            (6) Return a resolving promise from 'execute'
+                upon a success message from the checkout/processing endpoint
+
+*/
+
+export class Strategy implements PaymentStrategy {
+    protected _paymentProcessor?: PaymentProcessor;
+
+    constructor(
+        protected _store: CheckoutStore,
+        protected _orderActionCreator: OrderActionCreator
+    ) {}
+
+    async initialize(options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        if (!options || !options.methodId) {
+            // TODO: Can we make this object and attribute mandatory?
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        const { methodId } = options;
+
+        /*
+            TODO: probably too brittle to just use methodId going forward
+
+            if methodId is 'card', how do we differentiate between standard and "has own JS" implementations?
+
+            N.B. looking for a "has own JS" initializer, not finding one,
+            then falling back to a default one for that methodId
+            could produce some unpredictable (and likely broken) UX during execute
+        */
+        const initializer = initializers[methodId];
```

Comment:
rather than have a different range of initializer, maybe it's worth having different PPSDK strategies for each "type" of PPSDK payment.

We could have a `methodId: 'PPSDK'` and `submethodId: 'card'`. This way if methodId is PPSDK, it could have it's own `PPSDKStrategyRegistry` (if we foresee many different strategies will be required)

Reply:
I had started a strategies per initialisation type, much like you're suggesting. Then the argument was made for unifying them into a singular SDK strategy https://github.com/bigcommerce/checkout-sdk-js/pull/1108#discussion_r612884341 

As you point out, this does lead to a split then being needed at the 'initialiser' level of that singular strategy instead.

The downside being that the singular strategy is even more multifaceted.
The plus side being that we don't have a bunch of strategies that differ only at this initialisation step. 

I think it's something we'll need to keep a close eye on as we start adding real code and real tests.

You're right about the likely need for a registry either way 👍